sky/skylet/providers/lambda_cloud/node_provider.py

```diff
@@ -238,10 +253,21 @@ def create_node(self, node_config: Dict[str, Any], tags: Dict[str, str],
         # Create nodes
         instance_type = node_config['InstanceType']
         region = self.provider_config['region']
+
         if config_tags[TAG_RAY_NODE_KIND] == NODE_KIND_HEAD:
             name = f'{self.cluster_name}-head'
+            # Occasionally, the head node will continue running for a short
+            # period after termination. This can lead to the following bug:
+            #   1. Head node autodowns but continues running.
+            #   2. The next autodown event is triggered, which executes ray up.
+            #   3. Head node stops running.
+            # In this case, a new head node is created after the cluster has
+            # terminated. We avoid this with the following check:
+            if self.on_head:
+                raise lambda_utils.LambdaCloudError('Head already exists.')
```

Comment:
Should we consider exiting the skylet event or resetting the autodown config as well, to make sure this problem will not happen to the other clouds?

Reply:
Sounds good to me!